revolutionpi/encoder.go

```diff
@@ -0,0 +1,101 @@
+//go:build linux
+
+// Package revolutionpi implements the Revolution Pi board GPIO pins.
+package revolutionpi
+
+import (
+	"context"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"go.viam.com/rdk/components/encoder"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+)
+
+type revolutionPiEncoder struct {
+	resource.Named
+	resource.TriviallyReconfigurable
```

Comment:
This resource claims to be trivially reconfigurable, but the config describes the pin name. If someone changes the pin name in their config, aren't we supposed to do something during reconfiguration? I wonder if this line was supposed to be `resource.AlwaysRebuild` instead of `TriviallyReconfigurable`.